python-spec/src/somacore/query/query.py

```diff
@@ -241,12 +241,25 @@ def varp(self, layer: str) -> data.SparseRead:
         """
         return self._axisp_inner(_Axis.VAR, layer)
 
+    def obsm(self, layer: str) -> data.SparseRead:
+        """Returns an ``obsm`` layer as a sparse read.
+        Lifecycle: maturing
+        """
+        return self._axism_inner(_Axis.OBS, layer)
+
+    def varm(self, layer: str) -> data.SparseRead:
+        """Returns an ``varm`` layer as a sparse read.
+        Lifecycle: maturing
+        """
+        return self._axism_inner(_Axis.VAR, layer)
+
     def to_anndata(
         self,
         X_name: str,
         *,
         column_names: Optional[AxisColumnNames] = None,
         X_layers: Sequence[str] = (),
+        obsm_keys: Sequence[str] = [],
```

Comment:
Need to use consistent names for X and obsm:
* `X_layers`
* `obsm_keys`

Suggest simply using `_layers`

Also, use a tuple, not a list, as the default, a la `X_layers`

Reply:
Also, missing the rest of the layers (varm, etc).  Any reason we should not add all of it in one pass?

Reply:
Before I add the other layers, I'd like the approach to be validated

Reply:
By the way, it looks like AnnData uses ["keys"](https://anndata.readthedocs.io/en/latest/generated/anndata.AnnData.obsm.html) to define the members of obsm, etc. [layers](https://anndata.readthedocs.io/en/latest/generated/anndata.AnnData.layers.html) seems to specifically refer to X. I'll leave the naming as is right now, but open to more discussion.